### PR TITLE
Paul K Classification !

### PR DESCRIPTION
**Pull request by llpk79**

### CODE REVIEW

**Review by hughjafro — COMMENTED** [review]
Module 1:

- Start a clean notebook.
  - Make sure you use comments in your code
- Do train/validate/test split with the Tanzania Waterpumps data.
  - Explore different train_size and test_size
- Begin to explore and clean the data.
  - Import has correct shape
  - Good choice to evaluate value counts to further explore data size
  - Effective way to address zeroes and NaNs and methodical approach to analyze each feature
  - Ingenious to use .now for aging purposes
- Select different numeric and categorical features.
  - The Permit feature has a quite a few missing values
- Do one-hot encoding. (Remember it may not work with high cardinality categoricals.)
- Scale features.
- Use scikit-learn for logistic regression.
  - Check
- Get your validation accuracy score.
  - nice score
- Get and plot your coefficients.
  - Make sure to plot the coeff with something like
    - coefficients = pd.Series(model.coef_[0], X_train_encoded.columns)
    - plt.figure(figsize=(10, 10))
    - coefficients.sort_values().plot.barh();
- Submit your predictions to our Kaggle competition.
- Commit your notebook to your fork of the GitHub repo.

**Review by hughjafro — COMMENTED** [review]
**Module2 Decision Tree:**
- Make sure you try different algorithms
  - DecisionTree
  - RandomForest
  - XGBoost
- Hopefully, you are exploring various features
- Stretch: How does changing the train/test size affect your score?

**Comment by hughjafro** [comment]
### Module 03 Random Forest:

**Overall:** Great job getting into the 80's. You are 0.5% away from 2nd!!!!
**Feedback:** Comment your code. Try more feature engineering.

_Do more exploratory data analysis, data cleaning, feature engineering, and feature selection._
- Changing the features that are used and even creating new features may give you a performance boost. Would squaring certain features be justified?

_Try a Random Forest Classifier._
- I did not see this in your last commit, but I am sure that you have used this already.

_Try Ordinal Encoding._
- You may have done this in a later version, but this is worth attempting.

**Submit new predictions.**

**Comment by hughjafro** [comment]
### Module 04 Confusion Matrix:

**Overall:** Great job this week
**Feedback:** Very nice commented code. Your future self will thank you.

_Get and interpret the confusion matrix for classification models_
- At least take a stab at the confusion matrix
- What does the c-matrix tell you?
- What is the significance of the precision, recall, f-1 scores?

_Kaggle Challenge_
- Good job pushing yourself.
- Ask yourself, what you would do differently on your next Kaggle challenge? In approach and execution.

**Great Job**‼️
